buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -262,45 +262,47 @@ private void setDistributionType(ElasticsearchDistribution distribution, TestDis
         }
     }
 
-    @Override
-    public void plugin(RegularFileProperty plugin) {
-        this.plugins.add(plugin.map(p -> p.getAsFile().toURI()));
+    // package protected so only TestClustersAware can access
+    @Internal
+    List<Configuration> getPluginAndModuleConfigurations() {
+        return pluginAndModuleConfigurations;
     }
 
-    @Override
-    public void plugin(Provider<URI> plugin) {
-        requireNonNull(plugin, "Plugin name can't be null");
-        checkFrozen();
-        if (plugins.contains(plugin)) {
-            throw new TestClustersException("Plugin already configured for installation " + plugin);
+    // creates a configuration to depend on the given plugin project, then wraps that configuration
+    // to grab the zip as a file provider
+    private Provider<RegularFile> maybeCreatePluginOrModuleDependency(String path) {
+        Project depProject = project.project(path);
+        Configuration configuration = project.getConfigurations().findByName(depProject.getName());
+        if (configuration == null) {
+            configuration = project.getConfigurations().create(depProject.getName());
+            DependencyHandler deps = project.getDependencies();
+            deps.add(depProject.getName(), deps.project(Map.of("path", path, "configuration", "zip")));
         }
-        this.plugins.add(plugin);
+        pluginAndModuleConfigurations.add(configuration);
+        Provider<File> fileProvider = configuration.getElements().map(s -> s.stream().findFirst().get().getAsFile());
+        return project.getLayout().file(fileProvider);
```

Comment:
I don't think this extra wrapping step is necessary. The result of `findFirst()` should return a `RegularFile` although we'd have to cast it as such. If what we get _isn't_ a `RegularFIle` we should error, as that indicates something went wrong.

Reply:
The collection is of File, not RegularFile. The map method lambda gets a Set<File>, and since Set in java has no way to get the "first", we must stream it and grab the first.

Reply:
It's actually a `Set` of `FileSystemLocation`, which you them map to a `File` via `getAsFile()`. You then have to wrap the `File` again to get a `RegularFile` back. What I'm saying is that the `FileSystemLocation` implementation type actually _is_ `RegularFile` already, you just have to cast it, rather than to to a `File` then back to a `RegularFile`.